pkg/gpu/context.go

```diff
@@ -187,7 +187,14 @@ func (ctx *systemContext) getCudaSymbols(path string) (*symbolsEntry, error) {
 		return data, nil
 	}
 
-	data, err := cuda.GetSymbols(path)
+	wantedSmVersions := make(map[uint32]struct{})
+	for _, smVersion := range ctx.deviceSmVersions {
+		wantedSmVersions[uint32(smVersion)] = struct{}{}
+	}
```

Comment:
maybe better to calculate it once for each device instead of on each call to getCudaSymbols?